lib/scripts/collection_list.rb

```diff
@@ -0,0 +1,32 @@
+def collections_path(collection)
+  if collection.collections.empty?
+    collection_paths = [[{"title": collection.title, "id": collection.id}]]
+  else
+    collection_paths = []
+  end
+
+  collection.collections.each do |parent_collection|
+    recursive_collection_path(
+      parent_collection,
+      [{"title": parent_collection.title, "id": parent_collection.id},
+       {"title": collection.title, "id": collection.id}],
+      collection_paths
+    )
+  end
+
+  collection_paths
+end
+
+def recursive_collection_path(collection, path, collection_paths)
+  if collection.collections.empty?
+    collection_paths << path
+  else
+    collection.collections.each do |collection|
+      recursive_collection_path(
+        collection,
+        [{"title": collection.title, "id": collection.id}] + path,
+        collection_paths
+      )
+    end
+  end
+end
```

Comment:
I had some small changes that I think make this a little cleaner:

```
def collections_path(collection)
  if collection.collections.empty?
    collection_paths = [[name_describing_data(collection)]]
  else
    collection_paths = []
    path = [{name_describing_data(parent_collection)},
            {name_describing_data(collection)}]

    collection.collections.each do |parent_collection|
      recursive_collection_path(parent_collection, path, collection_paths)
    end

    collection_paths
  end

def recursive_collection_path(collection, path, collection_paths)
  if collection.collections.empty?
    collection_paths << path
  else
    collection.collections.each do |collection|
      path = [name_describing_data(collection)] + path
      recursive_collection_path(collection, path, collection_paths)
    end
  end
end

# I wasn't sure what this was exactly, so I just put a placeholder name
def name_describing_data(collection)
  {"title": collection.title, "id": collection.id}
end
```

Reply:
I meant less describing the data literally and more describing what data represents. It looks like it's being used in for collection paths, but I don't know what exactly it is in a collection path. Is it the current collections path or path fragment?